kornia/feature/adalam/core.py

```diff
@@ -2,13 +2,51 @@
 from typing import Optional, Tuple, Union
 
 import torch
+from typing_extensions import NotRequired, TypedDict
 
-from kornia.core import Tensor, concatenate, tensor, where
+from kornia.core import Device, Tensor, concatenate, tensor, where
 
 from .ransac import ransac
 from .utils import dist_matrix, orientation_diff
 
 
+class AdalamConfig(TypedDict):
```

Comment:
I think this is fine because it was already a dict, but maybe we should start adopting more dataclasses since it's more predictable and less error prone to access keys

https://dev.to/meeshkan/typeddict-vs-dataclasses-in-python-epic-typing-battle-onb

Reply:
Totally agree, I just tried to keep it closer than it already was here -- The more complex configs I didn't touch so as not to risk breaking anything in this patch. But I believe we can upgrade all of them to dataclasses

Reply:
i guess research guys love dicts, wondering if we can make a dict-dataclass to go forth and back

Reply:
/cc @ducha-aiki opinions?

Reply:
> to go forth and back

```python
>>> from dataclasses import dataclass, asdict
>>> @dataclass
... class A:
...     b: int
... 
>>> asdict(A(1))
{'b': 1}
>>> A(**asdict(A(1)))
A(b=1)
```

Reply:
nice